mitxgraders/plugins/integralgrader.py

```diff
@@ -338,42 +366,67 @@ def raw_check(self, answer, cleaned_input):
+            try:
+                expected_re, expected_im = self.evaluate_int(
+                    answer['integrand'],
+                    answer['lower'],
+                    answer['upper'],
+                    answer['integration_variable'],
+                    varscope=varscope,
+                    funcscope=funcscope
+                )
+            except IntegrationError as e:
+                msg = "Integration Error with author's stored answer: {}"
+                raise ConfigError(msg.format(e.message))
+
+            student_re, student_im = self.evaluate_int(
+                cleaned_input['integrand'],
+                cleaned_input['lower'],
+                cleaned_input['upper'],
+                cleaned_input['integration_variable'],
+                varscope=varscope,
+                funcscope=funcscope
+                )
 
             # scipy raises integration warnings when things go wrong,
             # except they aren't really warnings, they're just printed to stdout
             # so we use quad's full_output option to catch the messages, and then raise errors.
             # The 4th component only exists when its warning message is non-empty
-            if len(student) == 4:
-                raise IntegrationError(student[3])
-            if len(expected) == 4:
-                raise ConfigError(expected[3])
+            if len(student_re) == 4:
+                raise IntegrationError(student_re[3])
+            if len(expected_re) == 4:
+                raise ConfigError(expected_re[3])
+            if len(student_im) == 4:
+                raise IntegrationError(student_im[3])
+            if len(expected_im) == 4:
+                raise ConfigError(expected_im[3])
 
             self.log(self.debug_appendix_template.format(
                 samplenum=i,
                 variables=varscope,
-                student_eval=student[0],
-                student_error=student[1],
-                student_neval=student[2]['neval'],
-                author_eval=expected[0],
-                author_error=expected[1],
-                author_neval=expected[2]['neval'],
+                student_re_eval=student_re[0],
+                student_re_error=student_re[1],
+                student_re_neval=student_re[2]['neval'],
+                student_im_eval=student_im[0],
+                student_im_error=student_im[1],
+                student_im_neval=student_im[2]['neval'],
+                author_re_eval=expected_re[0],
+                author_re_error=expected_re[1],
+                author_re_neval=expected_re[2]['neval'],
+                author_im_eval=expected_im[0],
+                author_im_error=expected_im[1],
+                author_im_neval=expected_im[2]['neval'],
             ))
 
             # Check if expressions agree
-            if not within_tolerance(expected[0], student[0], self.config['tolerance']):
+            if not within_tolerance(expected_re[0], student_re[0], self.config['tolerance']):
```

Comment:
Note that within_tolerance can handle complex numbers. Perhaps make a check value for a single within_tolerance call, so you don't have to repeat the consequences?